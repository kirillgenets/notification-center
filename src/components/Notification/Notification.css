/* Notification */

.notification {
  border-bottom: 1px solid #e5e5e5;

  color: #666666;

  cursor: pointer;
  transition: 0.3s;
}

.notification:hover {
  opacity: 0.6;
}

.notification:active {
  opacity: 0.4;
}

.unread {
  background-color: #F3F6F9;
}

.category, .text, .date {
  padding: 25px 0;
}

.category {
  padding-left: 46px;

  position: relative;
}

.unread .category::before {
  content: '';

  width: 9px;
  height: 100%;

  position: absolute;
  left: 0;
  top: 0;
}

.CRIT::before {
  background-color: #CD3E3E;
}

.INFO::before {
  background-color: #4174F7;
}

.SUCCESS::before {
  background-color: #8BD244;
}

.WARN::before {
  background-color: #E3D552;
}

.ERROR::before {
  background-color: rgb(197, 97, 4);
}

.text {
  width: 825px;

  text-transform: capitalize;
}

.date {
  width: 200px;
  padding-right: 116px;
}
