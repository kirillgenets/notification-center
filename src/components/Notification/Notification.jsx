import React from 'react';
import PropTypes from 'prop-types';
import styles from './Notification.css';

const Notification = (props) => {
  const readStatus = props.isRead ? 'read' : 'unread';

  return (
    <tr
      className={[styles.notification, styles[readStatus]].join(' ')}
      onClick={props.onNotificationClick}
    >
      <td className={[styles.category, styles[props.category.toLowerCase()]].join(' ')}>
        {props.category}
      </td>
      <td className={styles.text}>
        {props.text}
      </td>
      <td className={styles.date}>
        {props.date}
      </td>
    </tr>
  );
};

Notification.propTypes = {
  category: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  isRead: PropTypes.bool,
  date: PropTypes.string,
  onNotificationClick: PropTypes.func,
};

Notification.defaultProps = {
  readStatus: 'unread',
};

export default Notification;
