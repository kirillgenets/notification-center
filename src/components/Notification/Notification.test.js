import React from 'react';
import renderer from 'react-test-renderer';
import Notification from './Notification';

it('Read notification renders correctly', () => {
  const tree = renderer
    .create(
      <Notification
        category="WARN"
        text="This Session Is Running In Sandbox Mode. Changes Will Not Be Saved"
        isRead={true}
      />
    )
    .toJSON();
  
    expect(tree).toMatchSnapshot();
});

it('Unread notification renders correctly', () => {
  const tree = renderer
    .create(
      <Notification
        category="WARN"
        text="This Session Is Running In Sandbox Mode. Changes Will Not Be Saved"
        isRead={false}
      />
    )
    .toJSON();
  
    expect(tree).toMatchSnapshot();
});
