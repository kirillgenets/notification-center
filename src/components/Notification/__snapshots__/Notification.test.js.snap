// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read notification renders correctly 1`] = `
<tr
  className=" "
>
  <td
    className=" "
  >
    WARN
  </td>
  <td>
    This Session Is Running In Sandbox Mode. Changes Will Not Be Saved
  </td>
  <td />
</tr>
`;

exports[`Unread notification renders correctly 1`] = `
<tr
  className=" "
>
  <td
    className=" "
  >
    WARN
  </td>
  <td>
    This Session Is Running In Sandbox Mode. Changes Will Not Be Saved
  </td>
  <td />
</tr>
`;
